op-node/rollup/derive/plasma_data_source.go

```diff
@@ -38,12 +51,31 @@ func (s *PlasmaDataSource) Next(ctx context.Context) (eth.Data, error) {
 		}
 	}
 	// use the commitment to fetch the input from the plasma DA provider.
-	resp, err := s.fetcher.GetInput(ctx, s.comm, s.id.Number)
-	if err != nil {
+	data, err := s.fetcher.GetInput(ctx, s.comm, s.id)
+	// GetInput may call for a reorg if the pipeline is stalled and the plasma DA manager
+	// continued syncing origins detached from the pipeline origin.
+	if errors.Is(err, plasma.ErrReorgRequired) {
+		// challenge for a new previously derived commitment expired.
+		return nil, NewResetError(err)
+	} else if errors.Is(err, plasma.ErrExpiredChallenge) {
+		// this commitment was challenged and the challenge expired.
+		s.log.Warn("challenge expired, skipping batch", "comm", fmt.Sprintf("%x", s.comm))
+		s.comm = nil
+		// skip the input
+		return s.Next(ctx)
+	} else if errors.Is(err, plasma.ErrMissingPastWindow) {
+		return nil, NewCriticalError(fmt.Errorf("data for comm %x not available: %w", s.comm, err))
+	} else if err != nil {
```

Comment:
It'd be nice to special case `plasma.ErrPendingChallenge`. Else where we have `NotEnoughData` which forces a retry without going through the `TemporaryError` path (& the temp error path will actually slow derivation down because it's trying to back off against network errors).

Reply:
`if errors.Is(err, plasma.ErrPendingChallenge)` we could return NotEnoughData or check if the error is `plasma.ErrPendingChallenge` further up the stack. This is not an EOF. When we see a pending challenge I think we want to keep stepping without slowing down, but if we return a TemporaryError, we will slow down the stepping.